cvmfs/network/download.cc

```diff
@@ -1365,55 +1427,66 @@ bool DownloadManager::VerifyAndFinalize(const int curl_error, JobInfo *info) {
     if (info->compressed()) {
       zlib::DecompressInit(info->GetZstreamPtr());
     }
-    SetRegularCache(info);
-
-    // Failure handling
-    bool switch_proxy = false;
-    bool switch_host = false;
-    switch (info->error_code()) {
-      case kFailBadData:
-        SetNocache(info);
-        break;
-      case kFailProxyResolve:
-      case kFailProxyHttp:
-        switch_proxy = true;
-        break;
-      case kFailHostResolve:
-      case kFailHostHttp:
-      case kFailHostAfterProxy:
-        switch_host = true;
-        break;
-      default:
-        if (IsProxyTransferError(info->error_code())) {
-          if (same_url_retry) {
-            Backoff(info);
-          } else {
-            switch_proxy = true;
-          }
-        } else if (IsHostTransferError(info->error_code())) {
-          if (same_url_retry) {
-            Backoff(info);
-          } else {
-            switch_host = true;
-          }
-        } else {
-          // No other errors expected when retrying
-          PANIC(NULL);
-        }
-    }
-    if (switch_proxy) {
-      ReleaseCredential(info);
-      SwitchProxy(info);
-      info->SetNumUsedProxies(info->num_used_proxies() + 1);
-      SetUrlOptions(info);
-    }
-    if (switch_host) {
+
+    if (sharding_policy_.UseCount() > 0) {  // sharding policy
```

Comment:
I'm surprised that we don't switch a proxy in the if block if the sharding policy is set. Is this an oversight or do we switch proxies somewhere else with sharding?

Reply:
the `failover_indefinitely_` expects only 1 proxy group and the failover to a different proxy within the group is handled in
https://github.com/cvmfs/cvmfs/blob/247f3dfd14c3888a5d42dd048b2ad2f0b25a1799/cvmfs/network/download.cc#L880-L881